### PR TITLE
Kctlread -- WIP DO NOT MERGE

### PR DESCRIPTION
**Pull request by SchwarzM**
Implementing additional cli commands using the kubernikus generated client to allow non ui interaction with the admin cluster.

### CODE REVIEW

File: pkg/cmd/kubernikusctl/cluster.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	kubernikus "github.com/sapcc/kubernikus/pkg/client/kubernikus_generated"
+	"github.com/sapcc/kubernikus/pkg/client/kubernikus_generated/operations"
+	"github.com/sapcc/kubernikus/pkg/cmd"
+	"github.com/sapcc/kubernikus/pkg/cmd/printers"
+	"github.com/spf13/cobra"
+	"net/url"
+	//	"github.com/spf13/pflag"
+	// "github.com/gophercloud/gophercloud/openstack/identity/v3/tokens"
```
Comment (BugRoger) [inline pkg/cmd/kubernikusctl/cluster.go]: delete

File: pkg/cmd/kubernikusctl/cluster.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+)
+
+type ClusterOptions struct {
+	auth *CredentialsOptions
```
Comment (BugRoger) [inline pkg/cmd/kubernikusctl/cluster.go]: We probably need to restructure this. Extract common `AuthOptions` for and reuse in all relevant command

File: pkg/cmd/kubernikusctl/cluster.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+				return nil
+			},
+		))
+	cmd.CheckError(err)
+	printme := make([]printers.Printable, len(ok.Payload))
+	for i, cluster := range ok.Payload {
+		tmp := cluster
+		printme[i] = tmp
+	}
+	cmd.CheckError(printers.PrintTable(printme))
```
Comment (BugRoger) [inline pkg/cmd/kubernikusctl/cluster.go]: Doesn't this just do a hard exit? I'd usually bubble the errors up and exit at a central place.

File: pkg/client/models/cluster_print.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/sapcc/kubernikus/pkg/cmd/printers"
+)
+
+func (c *Cluster) GetFormats() map[string]struct{} {
+	ret := map[string]struct{}{
+		"table": struct{}{},
```
Comment (BugRoger) [inline pkg/client/models/cluster_print.go]: Extract into constants

**Comment by SchwarzM** [comment]
close and reopen with updated code

**Pull request closed without merging**